shared/js/participantStore.js

```diff
@@ -119,12 +120,38 @@ loop.store.ParticipantStore = function() {
         }
       }
 
+      if (!updatedParticipant.hasNotifiedOfJoin &&
+          updatedParticipant.isHere &&
+          !this._isPresenceExpired({
+            localPingTime: updatedParticipant.localPingTime
+          })) {
+        this._sendJoinNotification(updatedParticipant.participantName, false);
+        updatedParticipant.hasNotifiedOfJoin = true;
+      } else if (updatedParticipant.hasNotifiedOfJoin &&
+                 !updatedParticipant.isHere) {
+        this._sendLeaveNotification(updatedParticipant.participantName, true);
+        updatedParticipant.hasNotifiedOfJoin = false;
+      }
+
       this.setStoreState({
         participants: this._storeState.participants.set(userId, updatedParticipant)
       });
     },
 
     /**
+     * Determines if Presence has expired
+     *
+     * @param actionData
+     * @returns {boolean}
+     */
+    _isPresenceExpired(actionData) {
+      if (Date.now() - actionData.localPingTime > PING_TIME) {
+       return true;
```

Comment:
Are we actually passing in `actionData`? It looks more like a `participant` but why not just the time?